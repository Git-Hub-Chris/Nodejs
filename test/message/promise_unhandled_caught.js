// Flags: --expose-gc
'use strict';

const common = require('../common');
// TODO: Remove when making this the default.
process.env.NODE_UNHANDLED_REJECTION = 'ERROR_ON_GC';

new Promise(function(res, rej) {
  throw new Error('Fail');
});

setTimeout(() => {
  global.gc();
}, 1);

process.on('uncaughtException', common.mustCall((err, fromPromise) => {
  console.log(`Caught ${err.message}; From promise: ${fromPromise}`);
}));

process.on('exit', (code) => process._rawDebug(`Exit code ${code}`));
