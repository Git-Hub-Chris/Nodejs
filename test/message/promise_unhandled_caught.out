Caught Fail; From promise: FROM_PROMISE
(node:*) UnhandledPromiseRejectionWarning: Error: Fail
    at *promise_unhandled_caught.js:*
    at new Promise (<anonymous>)
    at Object.<anonymous> (*promise_unhandled_caught.js:*)
    at *
    at *
    at *
    at *
    at *
    at *
    at *
(node:*) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
Exit code 0
